### PR TITLE
Feat/pimple one adapter

### PR DESCRIPTION
**Pull request by dominikzogg**
Hello @fabpot

I tried, and failed until now to create an adapter for pimple 1.1 to get most of the existing providers running on Silex 2.0. For that i renamed the tests, to prevent naming conflict.

I used the old class, refactored it and use the old test code. As you can see, there are still errors. And i am not sure if they are solvable. The biggest problem at the moment is that i did not found a clean solution (its far away of one) to adapt the static methods share and prototype.

Regards @dominikzogg

### CODE REVIEW

**Comment by dominikzogg** [comment]
https://github.com/silexphp/Silex/pull/1292

File: src/PimpleProtected.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+class PimpleProtected
```
Comment (stof) [inline src/PimpleProtected.php]: is this class really necessary ? See how it was done in the Silex legacy provider support

File: src/PimpleShared.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+class PimpleShared
```
Comment (stof) [inline src/PimpleShared.php]: the class should be marked internal.

and you may want to make it invokable to be more BC

**Comment by stof** [comment]
@dominikzogg is this PR actually needed now ?

**Comment by dominikzogg** [comment]
@stof There are providers as for example the doctrine ones (and some orm provider, thirdparty) which create new \Pimple instances, i see no way to handle it within silex itself. I think the better question is if we should try to make a bc layer @fabpot ?

File: src/Pimple.php [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+ *
+ * @package pimple
+ * @author  Fabien Potencier
+ */
+class Pimple implements ArrayAccess
+{
+    /**
+     * @var \Pimple\Container
+     */
+    protected $container;
```
Comment (stof) [inline src/Pimple.php]: should be private

File: src/Pimple.php [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+     * @param string $id       The unique identifier for the object
+     * @param callable $callable A service definition to extend the original
+     *
+     * @return Closure The wrapped closure
+     *
+     * @throws InvalidArgumentException if the identifier is not defined or not a service definition
+     */
+    public function extend($id, $callable)
+    {
+        return $this->container->extend($id, $callable);
```
Comment (stof) [inline src/Pimple.php]: does this match the old behavior or no ?

File: src/PimpleShared.php (hunk 2 above)
Comment (dominikzogg) [inline src/PimpleShared.php]: https://github.com/silexphp/Silex/pull/1292#issuecomment-164247282

File: src/Pimple.php (hunk 4 above)
Comment (dominikzogg) [inline src/Pimple.php]: https://github.com/silexphp/Silex/pull/1292#issuecomment-164247230

**Pull request closed without merging**